### ISSUE TITLE
False "array bound violated error" of two dimension arrays

### ISSUE THREAD

**Issue by guhetier** [esbmc/esbmc#2]
A failing assertion over a value on the second column of a two dimension array
raise the error `array bounds violated` instead of the assertion violation message.

A minimal example follows :

```c
/* esbmc esbmc_array_bound_err.c */
int main(void)
{ 
    int a = 1;
    int b = 0;
    int t[2][1]  = { {0}, {1}};
    __ESBMC_assert(t[a][b] == 0, "TEST1");
    return (0);
}
```

Error trace :
```
ESBMC version 4.2.0 64-bit x86_64 linux
file esbmc_array_bound_err.c: Parsing
Converting
Type-checking esbmc_array_bound_err
Generating GOTO Program
GOTO program creation time: 0.094s
GOTO program processing time: 0.002s
Starting Bounded Model Checking
Symex completed in: 0.003s (17 assignments)
Slicing time: 0.000s
Generated 5 VCC(s), 1 remaining after simplification
No solver specified; defaulting to Boolector
Encoding remaining VCC(s) using bit-vector arithmetic
Encoding to solver time: 0.000s
Solving with solver Boolector 2.4.1
Encoding to solver time: 0.000s
Runtime decision procedure: 0.002s
Building error trace

Counterexample:

State 1 file esbmc_array_bound_err.c line 7 function main thread 0
c::main
----------------------------------------------------
Violated property:
  file esbmc_array_bound_err.c line 7 function main
  array bounds violated: array `t'[] upper bound
  t[a][b] == 0

VERIFICATION FAILED
```

**Comment by mikhailramalho** [comment]
Hum, apparently esbmc finds the property violation and it's printing the wrong message. I'm looking into it now.

**Comment by mikhailramalho** [comment]
Fixed the issue. It will be released as part of ESBMC 4.2.2.

Thank you for the report.

**Comment by mikhailramalho** [comment]
Hi,

The fixes were released as part of ESBMC 4.3.
